js/src/contract/item.ts

```diff
@@ -1292,23 +326,10 @@ export async function getItemProgram(
   env: string,
   customRpcUrl: string
 ): Promise<ItemProgram> {
-  if (customRpcUrl) log.debug("USING CUSTOM URL", customRpcUrl);
-
-  const solConnection = new web3.Connection(customRpcUrl || getCluster(env));
-
-  if (anchorWallet instanceof web3.Keypair)
-    anchorWallet = new NodeWallet(anchorWallet);
-
-  const provider = new AnchorProvider(solConnection, anchorWallet, {
-    preflightCommitment: "recent",
-  });
-
-  const idl = await Program.fetchIdl(ITEM_ID, provider);
-
-  const program = new Program(idl, ITEM_ID, provider);
-
-  return new ItemProgram({
-    id: ITEM_ID,
-    program,
-  });
+  return Program.Program.getProgramWithWallet(
+    ItemProgram, // QUESTION - What should go here?
```

Comment:
@long-banana What should this be?

Reply:
Exactly what you have here is correct. This argument is an unfortunate side-effect of getting the return types to work when extending an abstract class such as Program. I chose to force the argument to be specified, because otherwise to get the types to work right, every time someone calls the Program helpers to build a program object they would have to add a cast (`as ItemProgram`) on the returned value. That seemed to me like something that would be easy for someone to forget rather than being explicit and forcing it to be passed in as an argument to allow the generic return type to work.

If you know how to work around this issue, I'd absolutely love to hear it! I've spent more time than I would have liked trying to figure out an alternative that both functioned well, and didn't cause additional burden on the caller.